modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/service/impl/ModelServiceImpl.java

```diff
@@ -70,9 +101,12 @@ public void markToRebuild(String modelUUID, String metricKey, Boolean toRebuild)
     @Override
     public void updateThresholds(String modelUUID, String metricKey, Map<String, Object> thresholds) {
 
+        AssertUtil.notNull(modelUUID, "Model UUID can't be null");
+        AssertUtil.notNull(thresholds, "Thresholds can't be null");
+
         Integer modelID = modelRepositoryCustom.getModelID(metricKey, modelUUID);
-        Model model = modelRepository.getModelById(modelID);
+        Model model = modelRepository.getOne(modelID);
```

Comment:
Ha, this is a great example of where my suggestion of just returning the `Model` would allow you to have a single call instead of two.

Also, you shouldn't be calling `modelRepositoryCustom`. All repo calls should go through the `ModelRepository` abstraction. You can wire custom implementation code into `ModelRepository` interface methods. See the Spring Data JPA docs.